Analysis/Tasks/PWGCF/FemtoDream/include/FemtoDream/FemtoDreamMath.h

```diff
@@ -0,0 +1,99 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file FemtoDreamMath.h
+/// \brief Definition of the FemtoDreamMath Container for math calculations of quantities related to pairs
+/// \author Valentina Mantovani Sarti, TU München, valentina.mantovani-sarti@ph.tum.de
+
+#ifndef ANALYSIS_TASKS_PWGCF_FEMTODREAM_FEMTODREAMMATH_H_
+#define ANALYSIS_TASKS_PWGCF_FEMTODREAM_FEMTODREAMMATH_H_
+
+#include "Math/Vector4D.h"
+#include "Math/Boost.h"
+#include "TLorentzVector.h"
+#include "TMath.h"
+
+#include <iostream>
+
+namespace o2::analysis
+{
+namespace femtoDream
+{
+
+/// \class FemtoDreamMath
+/// \brief Container for math calculations of quantities related to pairs
+
+class FemtoDreamMath
+{
+ public:
+
+  template <typename T>
+  static float getkstar(const T& part1, const float mass1, const T& part2, const float mass2)
+  {
+    ROOT::Math::PtEtaPhiMVector vecpart1(part1.pt(), part1.eta(), part1.phi(), mass1);
+    ROOT::Math::PtEtaPhiMVector vecpart2(part2.pt(), part2.eta(), part2.phi(), mass2);
+
+    ROOT::Math::PtEtaPhiMVector trackSum = vecpart1 + vecpart2;
+
+    float beta = trackSum.Beta();
+    float betax = beta * std::cos(trackSum.Phi()) * std::sin(trackSum.Theta());
+    float betay = beta * std::sin(trackSum.Phi()) * std::sin(trackSum.Theta());
+    float betaz = beta * std::cos(trackSum.Theta());
+
+    float px1 = vecpart1.pt() * std::cos(vecpart1.Phi());
+    float py1 = vecpart1.pt() * std::sin(vecpart1.Phi());
+    float pz1 = vecpart1.pt() * std::sinh(vecpart1.Eta());
+    float px2 = vecpart2.pt() * std::cos(vecpart2.Phi());
+    float py2 = vecpart2.pt() * std::sin(vecpart2.Phi());
+    float pz2 = vecpart2.pt() * std::sinh(vecpart2.Eta());
+
+    ROOT::Math::PxPyPzMVector PartOneCMS(px1, py1, pz1, vecpart1.M());
```

Comment:
one of the old comments must have gotten lost.
Can't you do here just ROOT::Math::PxPyPzMVector PartOneCMS(vecpart1);

Reply:
ah yes sorry this then needs to be a PtEtaPhiMVector, but that shouldn't matter